### PR TITLE
Join story, styles and UX

### PR DESCRIPTION
**Pull request by stuartromanek**
A take-two on #2343, this adds a storybook story for Joins, re works some style cruft and a little bit of UX .. works in both SB and apos 3 demo

### CODE REVIEW

File: modules/@apostrophecms/schema/ui/apos/components/AposInputJoin.vue [hunk 1]
```diff
@@ -69,6 +85,12 @@ export default {
         return { message: 'required' };
       }
 
+      // if the original status was disabled, no validation should change that
+      if (this.originalDisabled) {
```
Comment (abea) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputJoin.vue:89]: is this really an issue for all fields?

File: modules/@apostrophecms/schema/ui/apos/components/AposSearchList.vue [hunk 2]
```diff
@@ -72,20 +68,21 @@ export default {
 .item {
   display: flex;
   justify-content: space-between;
-  padding: 15px 25px;
+  margin: 10px;
+  padding: 10px 20px;
   letter-spacing: 0.5px;
-
+  border-radius: 5px;
```
Comment (abea) [inline modules/@apostrophecms/schema/ui/apos/components/AposSearchList.vue:74]: We have this stored as `--a-border-radius`.

**Review by abea — CHANGES_REQUESTED** [review]
A question and a minor thing.

File: modules/@apostrophecms/schema/ui/apos/components/AposInputJoin.vue (hunk 1 above)
Comment (stuartromanek) [reply modules/@apostrophecms/schema/ui/apos/components/AposInputJoin.vue:89]: I don't think so .. joins can be disabled by reaching their `max` limit, but then can be re-enabled by removing an item from the slat list. This two-ways-to-affect-state is unique to things that use the slat list.

A normal input is just going to be disabled from the get go and won't get re validated

**Comment by boutell** [comment]
@falkodev the renaming of joins to relationships has landed, so this will need a manual merge with the upstream work in 3.0.

**Comment by boutell** [comment]
@stuartromanek  the renaming of joins to relationships has landed, so this will need a manual merge with the upstream work in 3.0.

**Comment by stuartromanek** [comment]
updated to new Relationship name

**Review by abea — APPROVED** [review]

File: modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.stories.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        field: {
+          name: 'myJoin',
+          label: 'Link to A Page',
+          help: 'Choose a page from the joiner to link to it',
+          type: 'join',
+          withType: 'product',
+          max
+        },
+        value: {
+          data: getData(prepoulate)
```
Comment (boutell) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.stories.js:51]: prepopulate, not prepoulate

File: modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.stories.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    'Limit', {
+      None: null,
+      One: 1,
+      Two: 2,
+      Three: 3
+    },
+    null
+  );
+
+  const prepoulate = select(
```
Comment (boutell) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.stories.js:26]: typo

File: modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.vue [hunk 5]
```diff
@@ -44,17 +47,30 @@ export default {
...
   },
   mounted() {
     this.validateAndEmit();
   },
+  computed: {
+    pluralLabel() {
+      return apos.modules[this.field.withType].pluralLabel;
+    },
+    placeholder() {
+      return this.field.placeholder || `Search ${this.pluralLabel}`;
```
Comment (boutell) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.vue:68]: I think that if we're going to do this "auto-guess the verbiage" thing we should push it as config from the server side where i18n is easy. I would be open with that being a followup ticket though as long as it gets opened.

File: modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.vue [hunk 6]
```diff
@@ -44,17 +47,30 @@ export default {
...
+    },
+    browseLabel() {
+      return `Browse ${this.pluralLabel}`;
```
Comment (boutell) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.vue:71]: Same comment about the origin of browser-side text and i18n.

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.vue (hunk 5 above)
Comment (stuartromanek) [reply modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.vue:70]: opened an issue https://apostrophecms.atlassian.net/browse/A30U-436

File: modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.stories.js (hunk 4 above)
Comment (stuartromanek) [reply modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.stories.js]: fixec

**Review by boutell — APPROVED** [review]

**Pull request merged**